torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -216,21 +216,17 @@ void pushProfilingCallbacks() {
 std::string shapesToStr(const std::vector<std::vector<int64_t>>& shapes) {
   std::ostringstream oss;
   oss << "[";
-  for (auto t_idx = 0; t_idx < shapes.size(); ++t_idx) {
-    if (t_idx > 0) {
-      oss << ", ";
-    }
+  for (const auto& shape : shapes) {
     oss << "[";
-    for (auto s_idx = 0; s_idx < shapes[t_idx].size(); ++s_idx) {
-      if (s_idx > 0) {
-        oss << ", ";
-      }
-      oss << shapes[t_idx][s_idx];
-    }
-    oss << "]";
+    std::copy(shape.begin(), shape.end(), std::ostream_iterator<int64_t>(oss, ", "));
+    oss.seekp(-2, std::ios_base::end);
+    oss << "], ";
   }
+  oss.seekp(-2, std::ios_base::end);
   oss << "]";
-  return oss.str();
+  auto rc = oss.str();
+  rc.pop_back();
```

Comment:
is it correct to do it here? we just outputted that last ']" above

Reply:
If you are ok to have trailing space in the string, then it's fine, as `stringstream::seekp` does not delete symbols from the stream.